server/handlers/chat.go

```diff
@@ -15,17 +17,39 @@ func (s *Server) handleChat(w http.ResponseWriter, r *http.Request) {
 	}
 	defer conn.Close()
 
+	// Read token payload with username
+	verifiedToken := r.Context().Value(ContextKey).(*jwt.Token)
+
+	subject, tokenErr := verifiedToken.Claims.GetSubject()
+	if tokenErr != nil {
+		respondWithError(w, "Invalid Token", http.StatusUnauthorized)
+		return
+	}
+
+	// Subscribe to Redis channel and receive messages for user
+	go data.JoinChannel(conn, subject)
+
 	for {
-		m, msg, err := conn.ReadMessage()
-		if err != nil {
-			fmt.Println("error reading message:", err)
+		chatMessage := data.ChatMessage{Type: "message"}
+
+		if err := conn.ReadJSON(&chatMessage); err != nil {
+			fmt.Println("Bad Request,", err.Error())
 			continue
 		}
 
-		err = conn.WriteMessage(m, msg)
-		if err != nil {
-			fmt.Println("error writing message:", err)
+		if verifyErr := verifyMessage(&chatMessage, subject); verifyErr != nil {
+			fmt.Println(verifyErr.Error())
 			continue
 		}
+
+		data.SendMessage(chatMessage)
 	}
 }
+
+func verifyMessage(message *data.ChatMessage, username string) error {
+	if message.From != username {
+		return fmt.Errorf("incoming ChatMessage \"from\" field does not match username")
```

Comment:
nitpick: You're not formatting anything, so no need to use the `fmt` package. Why not use the `errors` package? 
https://pkg.go.dev/errors#New